components/form/form-helper.js

```diff
@@ -152,6 +153,12 @@ const _hasFormData = (node, submitter) => {
 			return false;
 		}
 	}
+	if (tagName === 'd2l-input-checkbox') {
```

Comment:
I don't love `form-helper` being aware of checkbox here, although it's already aware of `<input type="checkbox">` above to apply the same logic. I was hoping that `this.setFormValue(undefined)` when unchecked would accomplish this, but it submits `null` as the value. I'm worried about "fixing" that behaviour.

This also acts as a feature flag for the change, albeit in a creative way. I originally implemented the flag similar to how @dbatiste did the tooltip popover stuff -- by duplicating the whole class and then conditionally registering one or the other. But since files aren't reloaded during tests, there wasn't a good way to unit test that.